Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/ProgressIndicators/ProgressIndicatorOrbsRotator.cs

```diff
@@ -0,0 +1,207 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Threading.Tasks;
+using TMPro;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Extensions.SceneTransitions
+{
+    /// <summary>
+    /// This class manages the 'rotating circle of dots' effect
+    /// that is used as a Progress Indicator effect.
+    /// </summary>
+    public class ProgressIndicatorOrbsRotator : MonoBehaviour, IProgressIndicator
+    {
+        public Transform MainTransform { get { return transform; } }
+        public ProgressIndicatorState State { get { return state; } }
+        public float Progress { set { progress = value; } }
+        public string Message { set { messageText.text = value; } }
+
+        [SerializeField]
+        private GameObject[] orbs = null;
+        
+        [SerializeField]
+        private TextMeshPro messageText = null;
+
+        [SerializeField]
+        private ProgressIndicatorState state = ProgressIndicatorState.Closed;
+        
+        public float RotationSpeedRawDegrees;
+        public float SpacingDegrees;
```

Comment:
For public fields, please add doccomments to explain what these are doing, even if they are reasonably named if they are here (i.e. spacing degrees - I assume these are degrees between the orbs, but maybe it's something else? - the fact that I have to make an assumption and can only validate it via looking at the implementation means having a doccomment would be helpful